index.js

```diff
@@ -246,313 +214,122 @@ module.exports.update = function(tags, filebuffer, options, fn) {
-            fn(err)
-            return
-        }
+function makeSwapParameters(fn) {
+    return (a, b) => fn(b, a)
+}
 
-        fs.writeFile(filepath, newData, 'binary', function(err) {
-            if(err) {
-                fn(err)
+// The reorderParameter is a workaround because the callback function
+// does not have a consistent interface between all the API functions.
+// Ideally, all the functions should align with the promise style and
+// always have the result first and the error second.
+// Changing this would break the current public API.
+// This could be changed internally and swap the parameter in a light
+// wrapper when creating the public interface and then remove in a
+// version 1.0 later with an API breaking change.
+function makePromise(
+    fn,
+    reorderParameters = fn => (a, b) => fn(a, b)
+) {
+    return new Promise((resolve, reject) => {
+        fn(reorderParameters((error, result) => {
+            if(error) {
+                reject(error)
             } else {
-                fn(false)
+                resolve(result)
             }
-        })
-    }.bind(this))
+        }))
+    })
 }
 
-module.exports.Promise = {
-    write: (tags, file) => {
-        return new Promise((resolve, reject) => {
-            this.write(tags, file, (err, ret) => {
-                if(err) {
-                    reject(err)
-                } else {
-                    resolve(ret)
-                }
-            })
-        })
-    },
-    update: (tags, file) => {
-        return new Promise((resolve, reject) => {
-            this.update(tags, file, (err, ret) => {
-                if(err) {
-                    reject(err)
-                } else {
-                    resolve(ret)
-                }
-            })
-        })
-    },
-    create: (tags) => {
-        return new Promise((resolve) => {
-            this.create(tags, (buffer) => {
-                resolve(buffer)
-            })
-        })
-    },
-    read: (file, options) => {
-        return new Promise((resolve, reject) => {
-            this.read(file, options, (err, ret) => {
-                if(err) {
-                    reject(err)
-                } else {
-                    resolve(ret)
-                }
-            })
-        })
-    },
-    removeTags: (filepath) => {
-        return new Promise((resolve, reject) => {
-            this.removeTags(filepath, (err) => {
-                if(err) {
-                    reject(err)
-                } else {
-                    resolve()
-                }
-            })
-        })
-    }
+const PromiseExport = {
+    create: (tags) => makePromise(create.bind(null, tags), makeSwapParameters),
+    write: (tags, file) => makePromise(write.bind(null, tags, file)),
+    update: (tags, file) => makePromise(update.bind(null, tags, file)),
+    read: (file, options) => makePromise(read.bind(null, file, options)),
+    removeTags: (filepath) => makePromise(removeTags.bind(null, filepath))

```

Comment:
Interestingly Scrutinizer has a false positive here and tag these `bind`s as unnecessary because there is no `this` parameters, but completely miss the points that more parameters are bound!

Reply:
I've reported it as a bug and got as reply: `Hi Jan, thanks for the report. We already had an internal issue for this, I've added your report to it and will keep you updated about its progress. If you need anything else, just reply to this ticket. `

If they don't fix it soon, I can just hide this issue.

Reply:
Excellent to have reported it, thanks.